kunta-api-server/src/main/java/fi/otavanopisto/kuntaapi/server/integrations/mwp/MwpNewsProvider.java

```diff
@@ -0,0 +1,333 @@
+    if (featuredMediaId == null) {
+      return null;
+    }
+    
+    ApiResponse<fi.otavanopisto.mwp.client.model.Attachment> response = mwpApi.getApi().wpV2MediaIdGet(String.valueOf(featuredMediaId), null);
+    if (!response.isOk()) {
+      logger.severe(String.format("Finding media failed on [%d] %s", response.getStatus(), response.getMessage()));
+    } else {
+      return response.getResponse();
+    }
+    
+    return null;
+  }
+
+  private Post findPostByArticleId(NewsArticleId newsArticleId) {
+    NewsArticleId kuntaApiId = idController.translateNewsArticleId(newsArticleId, MwpConsts.IDENTIFIER_NAME);
+    if (kuntaApiId == null) {
+      logger.severe(String.format("Failed to convert %s into MWP id", newsArticleId.toString()));
+      return null;
+    }
+    
+    ApiResponse<Post> response = mwpApi.getApi().wpV2PostsIdGet(kuntaApiId.getId(), null);
+    if (!response.isOk()) {
+      logger.severe(String.format("Finding post failed on [%d] %s", response.getStatus(), response.getMessage()));
+    } else {
+      return response.getResponse();
+    }
+    
+    return null;
+  }
+
+  private List<NewsArticle> translateNewsArticles(List<Post> posts) {
+    List<NewsArticle> result = new ArrayList<>();
+    
+    for (Post post : posts) {
+      result.add(translateNewsArticle(post));
+    }
+    
+    return result;
+  }
+
+  private NewsArticle translateNewsArticle(Post post) {
+    NewsArticle newsArticle = new NewsArticle();
+    
+    NewsArticleId postId = new NewsArticleId(MwpConsts.IDENTIFIER_NAME, String.valueOf(post.getId()));
+    NewsArticleId kuntaApiId = idController.translateNewsArticleId(postId, KuntaApiConsts.IDENTIFIER_NAME);
+    if (kuntaApiId == null) {
+      logger.info(String.format("Found new news article %d", post.getId()));
+      Identifier newIdentifier = identifierController.createIdentifier(IdType.NEWS_ARTICLE, MwpConsts.IDENTIFIER_NAME, String.valueOf(post.getId()));
+      kuntaApiId = new NewsArticleId(KuntaApiConsts.IDENTIFIER_NAME, newIdentifier.getKuntaApiId());
+    }
+    
+    newsArticle.setAbstract(post.getExcerpt().getRendered());
+    newsArticle.setContents(post.getContent().getRendered());
+    newsArticle.setId(kuntaApiId.getId());
+    newsArticle.setPublished(toOffsetDateTime(post.getDate()));
+    newsArticle.setTitle(post.getTitle().getRendered());
+    
+    return newsArticle;
+  }
+  
+  private Attachment translateAttachment(fi.otavanopisto.mwp.client.model.Attachment featuredMedia) {
+    Integer size = binaryHttpClient.getDownloadSize(featuredMedia.getSourceUrl());
+    AttachmentId id = getImageAttachmentId(featuredMedia.getId());
+    Attachment attachment = new Attachment();
+    attachment.setContentType(featuredMedia.getMimeType());
+    attachment.setId(id.getId());
+    attachment.setSize(size != null ? size.longValue() : null);
+    return attachment;
+  }
+  
+  private OffsetDateTime toOffsetDateTime(LocalDateTime date) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "toOffsetDateTime" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)